panel/panel.py

```diff
@@ -118,9 +97,13 @@ def ensure_panel(window: sublime.Window):
     return get_panel(window) or create_panel(window)
 
 
-def filter_errors(window, errors):
-    vids = [v.id() for v in window.views()]
-    return {vid: d for vid, d in errors.items() if vid in vids}
+def sort_errors(errors):
+    return sorted(errors, key=lambda x: (x["line"], x["start"], x["end"]))
+
+
+def get_window_raw_errors(window, errors):
+    bids = {v.buffer_id() for v in window.views()}
+    return {bid: sort_errors(d) for bid, d in errors.items() if bid in bids}
```

Comment:
I always thought about this as 

```
{bid: sorted(errors[bid]) for bid in {v.buffer_id() for v in window.views()}}

```

T.i. you don't iterate over `items()` and filter this via if bid in bids

Reply:
Yeah, makes sense to iterate over buffers in active window = shorter.